src/context/context.service.ts

```diff
@@ -94,8 +94,24 @@ export class ContextService {
     return $('<div>').html($.html()).text().trim();
   }
 
-  setHtmlBody(body: string): void {
-    this.cheerioBody = cheerio.load(body);
+     setHtmlBody(
+    body: string,
+    loadAsDocument = false,
```

Comment:
I will keep your suggestion while prefer to `loadAsDocument = true` by default to keep compatibility with former usage in renderPage and renderSlides.
However, it may be easier and obvious just to define two utilities to load the body in the context module:
- `setHtmlBody` as it was with `loadAsDocument = true` by default
- `setBody` with loadAsDocument = false` for the needed behaviour in the API